### PR TITLE
new style class and new style format string -> py3k

### PR DESCRIPTION
**Pull request by jhoonb**
Python 3.
New style class: https://docs.python.org/2/reference/datamodel.html#new-style-and-classic-classes
New format string: https://docs.python.org/3/library/string.html#format-examples

### CODE REVIEW

File: ddc_client.py [hunk 1]
```diff
@@ -59,9 +59,9 @@ def start(self,):
           # upgrade components if necessary
           need_restart = False
           for xml_upgrade in xml_response.findall("upgrades/upgrade"):
-            type = xml_upgrade.get("type")
+            ttype = xml_upgrade.get("type")
```
Comment (nilnilnil) [inline ddc_client.py:62]: What's the reason for the name change?

Comment (zekiel) [inline ddc_client.py:62]: ping @jhoonb  :) we can close this out or address  //cc @moollaza

**Comment by angiehilem** [comment]
Reaching out via email right now.

File: ddc_client.py (hunk 1 above)
Comment (jhoonb) [inline ddc_client.py:62]: @nilnilnil "type" is a built-in functions  https://docs.python.org/3/library/functions.html#type

**Comment by nilnilnil** [comment]
@angiehilem @jhoonb any reason you closed? I'm happy to be assigned if this is reopened because the changes still seem to stand.

Is it worth moving to Python 3 instead of Python 2.7? If so, why?

**Pull request closed without merging**